library/core/class.session.php

```diff
@@ -213,6 +213,24 @@ public function getCookie($Suffix, $Default = null) {
     }
 
     /**
+     * Return the time zone for the current user.
+     *
+     * @return string A valid IANA time zone identifier.
+     */
+    public function getTimeZone() {
+        $timeZone = $this->getAttribute('TimeZone');
+        $hourOffset = $this->hourOffset();
+
+        if ($timeZone) {
+            return $timeZone;
+        } elseif (is_int($hourOffset)) {
+            return "Etc/GMT/{$hourOffset}";
```

Comment:
The our offset needs to be passed through `sprintf('%+d', $hourOffset)` so that the "+" sign is captured.